webofneeds/won-node/src/main/java/won/node/camel/processor/general/ReferenceToUnreferencedMessageAddingProcessor.java

```diff
@@ -95,7 +97,7 @@ public WonMessage process(final WonMessage message) throws WonMessageProcessingE
     }
     //persist the message
     messageEventRepository.save(messageEventPlaceholders);
-    return new WonMessage(messageDataset);
+    return message;
```

Comment:
This is intended, and it's not really safe. However, I gave up the concept of immutable WonMessage objects for the current implementation as it creates a big management overhead. This way, it became easy to add properties to a newly added envelope (see also changes to WonMessageRoutes) one by one, and be sure that only one envelope is added as the message passes through the system.